### PR TITLE
[Remove] Types from GET/MGET

### PR DESCRIPTION
**Pull request by nknize**
This PR removes type support from the get and mget transport and rest actions along
with removal from ShardGetService.

relates #1940

### CODE REVIEW

**Comment by nknize** [comment]
Another docker failure:

```
> Task :distribution:docker:buildDockerImage FAILED
Exec output and error:
| Output for docker:Sending build context to Docker daemon  223.8MB
```

Refiring!

**Comment by nknize** [comment]
start gradle check

**Comment by nknize** [comment]
Non-reproducible random disconnect error:

```
./gradlew ':server:internalClusterTest' --tests "org.opensearch.cluster.ClusterHealthIT.testHealthOnMasterFailover" -Dtests.seed=5222100ABDBA97B0 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=es-PA -Dtests.timezone=Canada/Central -Druntime.java=17
```

```
org.opensearch.cluster.ClusterHealthIT > testHealthOnMasterFailover FAILED
    java.util.concurrent.ExecutionException: MasterNotDiscoveredException[NodeDisconnectedException[[node_s1][127.0.0.1:46577][cluster:monitor/health] disconnected]]; nested: NodeDisconnectedException[[node_s1][127.0.0.1:46577][cluster:monitor/health] disconnected];
        at __randomizedtesting.SeedInfo.seed([5222100ABDBA97B0:CCCD3F27BE508B56]:0)
        at org.opensearch.common.util.concurrent.BaseFuture$Sync.getValue(BaseFuture.java:281)
        at org.opensearch.common.util.concurrent.BaseFuture$Sync.get(BaseFuture.java:268)
        at org.opensearch.common.util.concurrent.BaseFuture.get(BaseFuture.java:99)
        at org.opensearch.cluster.ClusterHealthIT.testHealthOnMasterFailover(ClusterHealthIT.java:393)
```

Refiring!

**Comment by nknize** [comment]
start gradle check

**Comment by nknize** [comment]
ANOTHER non reproducible flaky test failure already reported in #1746 

```
REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.gateway.RecoveryFromGatewayIT.testReuseInFileBasedPeerRecovery" -Dtests.seed=5D9018D9DB3A08DD -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=ar-DZ -Dtests.timezone=America/Danmarkshavn -Druntime.java=17
```

```
org.opensearch.gateway.RecoveryFromGatewayIT > testReuseInFileBasedPeerRecovery FAILED
    java.lang.AssertionError: shard [test][0] on node [node_t1] has pending operations:
     --> RetentionLeaseBackgroundSyncAction.Request{retentionLeases=RetentionLeases{primaryTerm=1, version=1471, leases={peer_recovery/4ONFp4MdSgOm-e_3qIW5Fg=RetentionLease{id='peer_recovery/4ONFp4MdSgOm-e_3qIW5Fg', retainingSequenceNumber=846, timestamp=1645245603893, source='peer recovery'}, peer_recovery/J634nQ0wRaipZGI9GjlOZA=RetentionLease{id='peer_recovery/J634nQ0wRaipZGI9GjlOZA', retainingSequenceNumber=846, timestamp=1645245603893, source='peer recovery'}}}, shardId=[test][0], timeout=1m, index='test', waitForActiveShards=0}
    	at org.opensearch.index.shard.IndexShardOperationPermits.acquire(IndexShardOperationPermits.java:248)
    	at org.opensearch.index.shard.IndexShard.acquirePrimaryOperationPermit(IndexShard.java:3224)
    	at org.opensearch.action.support.replication.TransportReplicationAction.acquirePrimaryOperationPermit(TransportReplicationAction.java:1116)
    	at org.opensearch.action.support.replication.TransportReplicationAction$AsyncPrimaryAction.doRun(TransportReplicationAction.java:433)
```

**Comment by nknize** [comment]
start gradle check

**Review by dreamer-89 — APPROVED** [review]

File: client/rest-high-level/src/test/java/org/opensearch/client/CrudIT.java [hunk 1]
```diff
@@ -456,25 +447,7 @@ public void testMultiGetWithTypes() throws IOException {
...
-        GetResponse firstResponse = response.getResponses()[0].getResponse();
-        assertEquals("index", firstResponse.getIndex());
-        assertEquals("type", firstResponse.getType());
-        assertEquals("id1", firstResponse.getId());
-
-        GetResponse secondResponse = response.getResponses()[1].getResponse();
-        assertEquals("index", secondResponse.getIndex());
-        assertEquals("type", secondResponse.getType());
-        assertEquals("id2", secondResponse.getId());
+        multiGetRequest.add("index", "id2");
```
Comment (reta) [inline client/rest-high-level/src/test/java/org/opensearch/client/CrudIT.java:450]: Minor, but the test name  `testMultiGetWithTypes()` is not reflecting anymore what the test is doing, `testMultiGetWithIds()` ?

Comment (dblock) [reply client/rest-high-level/src/test/java/org/opensearch/client/CrudIT.java:450]: @nknize ^

**Pull request merged**